src/wallet/rpcwallet.cpp

```diff
@@ -3790,9 +3790,17 @@ static UniValue getaddressesbylabel(const JSONRPCRequest& request)
 
     // Find all addresses that have the given label
     UniValue ret(UniValue::VOBJ);
+    std::set<std::string> addresses;
     for (const std::pair<const CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
         if (item.second.name == label) {
-            ret.pushKV(EncodeDestination(item.first), AddressBookDataToJSON(item.second, false));
+            std::string address = EncodeDestination(item.first);
+            if (addresses.emplace(address).second) {
```

Comment:
It would be good to have a comment like "Should always be true" if this is just a defensive check for a theoretical thing that should never happen. Or if there's an actual case where this could be false it would be good to note an example of what could cause this. Also if this can be false, it would be good to note the change in behavior in the commit message, since this code now keeps the first dup entry instead of the last dup entry.

Reply:
> Or if there's an actual case where this could be false it would be good to note an example of what could cause this.

Not really sure, maybe @MeshCollider can help?

> since this code now keeps the first dup entry instead of the last dup entry.

I've mentioned above https://github.com/bitcoin/bitcoin/pull/15463#discussion_r259655966 that the behavior could be preserved.

Reply:
No need to go down a rabbit hole, but the code here is surprising, and adding a simple comment saying what the intention is (whatever it is) would make be an improvement, in my opinion. Would suggest: "mapAddressBook is not expected to contain duplicate address strings, but build a separate set as a precaution just in case it does." or "It is unclear whether mapAddressBook may contain duplicate address strings, so build a separate set as a precaution just in case it does."

Reply:
I can add the comment, but I'd love to know if duplicate it could hit duplicate keys.

Reply:
I don't think its possible to have duplicates, I think it should always be true in theory

Reply:
Comment added, thanks @ryanofsky.

Also changed to `assert()`, after all duplicate addresses is unexpected.